owa/app/js/components/manage_patient_identifier_sources/edit_patient_identifier_sources/localIdentifierGenerator.jsx

```diff
@@ -0,0 +1,144 @@
+import React, {Component} from 'react';
+import {Table, Button, Input} from 'reactstrap';
+import apiCall from '../../../utilities/apiHelper';
+import './editPatientIdentifierSources.css'
+
+export default class EditLocalIdentifierGenerator extends Component {
+
+  constructor(props){
+    super(props);
+    this.state = {
+      index: props.identifierSource.index,
+      initialized: props.identifierSource.initialized,
+      uuid: props.identifierSource.uuid,
+      name: props.identifierSource.name,
+      description: props.identifierSource.description,
+      identifierTypeValidator: props.identifierSource.identifierTypeValidator,
+      baseCharacterSet: props.identifierSource.baseCharacterSet,
+      firstIdentifierBase: props.identifierSource.firstIdentifierBase,
+      prefix: props.identifierSource.prefix,
+      suffix: props.identifierSource.suffix,
+      minLength: props.identifierSource.minLength,
+      maxLength: props.identifierSource.maxLength
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleFirstIdentifierBase(){
+    if(this.state.initialized){
+      return(this.state.firstIdentifierBase + " In Use - Unable to Modify ")
+    }
+    return(<Input 
+    id="firstIdentifierBase" 
+    value={this.state.firstIdentifierBase == null ? "" : this.state.firstIdentifierBase} 
+    onChange={event => this.handleChange(event)} 
```

Comment:
Since you have bind the method to the constructor, you can do `onChange={this.handleChange}` without passing in the event and it will still work perfectly well. This is cool though

Reply:
true

Reply:
fixed